src/tmpfiles/tmpfiles.c

```diff
@@ -2261,12 +2310,64 @@ static int remove_item(Item *i) {
         }
 }
 
+static int age_by_format(AgeBy age_by, char **ret_age_by_fmt) {
+        _cleanup_free_ char *tmp = NULL;
+        _cleanup_strv_free_ char **split = NULL;
+        char types[5] = { 0 };
+        char *p = NULL, *q = NULL;
+
+        assert(ret_age_by_fmt);
+
+        if (age_by == AGE_BY_UNSET) {
+                *ret_age_by_fmt = strdup("");
+                if (!*ret_age_by_fmt)
+                        return -ENOMEM;
+
+                return 0;
+        }
+
+        p = types;
+        if (age_by & AGE_BY_ATIME)
+                *p++ = 'a';
+
+        if (age_by & AGE_BY_BTIME)
+                *p++ = 'b';
+
+        if (age_by & AGE_BY_CTIME)
+                *p++ = 'c';
+
+        if (age_by & AGE_BY_MTIME)
+                *p++ = 'm';
+
+        /* Stop and rewind back to the beginning. */
+        *p = 0;
+        p = types;
+
+        do {
+                if (asprintf(&q, "%ctime", *p) < 0)
+                        return -ENOMEM;
+
+                if (strv_push(&split, q) < 0)
+                        return -ENOMEM;
```

Comment:
q is leaked on failure

Reply:
Oh, yes. But adding a `_cleanup_free` for `q` would result in a double free when `split` is being cleared. Will see if there's a better way.

Reply:
you can move the variable declartion into the inner scope?

Reply:
Would that help? I was thinking of calling `free` if `strv_push` failed.

Reply:
there's also strv_consume

Reply:
Perfect, that could work!